src/main/webapp/script.js

```diff
@@ -1,13 +1,333 @@
-/**
- * Adds a random location to the page.
- */
-function addRandomLocation() {  // eslint-disable-line no-unused-vars
-  const locations = ['Canada', 'Mexico', 'US'];
-
-  // Pick a random location.
-  const location = locations[Math.floor(Math.random() * locations.length)];
-
-  // Add it to the page.
-  const locationContainer = document.getElementById('location-container');
-  locationContainer.innerText = location;
+/** *************************
+Autocomplete Searchbar
+***************************/
+function autocomplete(inp, arr) {
+  // Create variable for the item on list currently focused on
+  let currentFocus;
+
+  // Function for when the user types into the searchbar
+  inp.addEventListener('input', function() {
+    let a; let b; let i; const val = this.value;
+
+    // Close any open autocomplete lists
+    closeAllLists();
+    if (!val) {
+      return false;
+    }
+
+    currentFocus = -1;
+
+    // Create a DIV element that will contain the items (values)
+    a = document.createElement('DIV');
+    a.setAttribute('id', this.id + 'autocomplete-list');
+    a.setAttribute('class', 'autocomplete-items');
+
+    // Append the DIV element as a child of the autocomplete container
+    this.parentNode.appendChild(a);
+
+    // Check if the item starts with the same letters as the text field value
+    for (i = 0; i < arr.length; i++) {
+      if (arr[i].substr(0, val.length).toUpperCase() == val.toUpperCase()) {
+        b = document.createElement('DIV');
+        b.innerHTML = '<strong>' + arr[i].substr(0, val.length) + '</strong>';
+        b.innerHTML += arr[i].substr(val.length);
+        b.innerHTML += '<input type=\'hidden\' value=\'' + arr[i] + '\'>';
+
+        // Execute a function when someone clicks on the item value (DIV element)
+        b.addEventListener('click', function(e) {
+          inp.value = this.getElementsByTagName('input')[0].value;
+          closeAllLists();
+        });
+
+        a.appendChild(b);
+      }
+    }
```

Comment:
1. Instead of `arr[i].substr(0, val.length).toUpperCase() == val.toUpperCase()` you can write `arr[i].toUpperCase().contains(val.toUpperCase())`

1. If `arr[i]` starts with `val`, then `arr[i].substr(0, val.length)` is the same as simply `val`.

1. This can be rewritten using [`Array.filter()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and [`Array.forEach()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach):
```suggestion
    // Check if the item starts with the same letters as the text field value
    arr.filter(c => c.toUpperCase().startsWith(val.toUpperCase()).forEach(c => {
      b = document.createElement('div');
      b.innerHTML = '<strong>' + val + '</strong>';
      b.innerHTML += c.substr(val.length);
      b.innerHTML += '<input type=\'hidden\' value=\'' + c + '\'>';

      // Execute a function when someone clicks on the item value (div element)
      b.addEventListener('click', function(e) {
        // use suggestion as input value and clear autocomplete list.
        inp.value = this.getElementsByTagName('input')[0].value;
        a.innerHTML = "";
      });

      a.appendChild(b);
    }
```

Reply:
Thank you for the suggestions! For some reason, when I used `Array.filter()` and `Array.forEach()`, there seems to be an unexplained error (there is a red line under the `}` on line 44).

![image](https://user-images.githubusercontent.com/60808509/80059952-553de980-84fb-11ea-8bfd-5d26469b818d.png)

Reply:
Hmm, that seems like it might just be a mismatched curly brace or parentheses somewhere

Reply:
Ah, it is a mismatched parentheses. The open parenthese on line 29 of my suggested change needs to be closed (probably on line 43 after the `}`)